Set up XDG_CONFIG_DIRS lookup in order:
1. custom XDG_CONFIG_DIRS from env
2. kdeprefix xdg config dirs
3. /etc/xdg

Index: kdecore/kernel/kstandarddirs.cpp
===================================================================
--- a/kdecore/kernel/kstandarddirs.cpp	(revision 1014759)
+++ b/kdecore/kernel/kstandarddirs.cpp	(working copy)
@@ -1664,16 +1664,13 @@
     {
         tokenize(xdgdirList, xdgdirs, QString(QChar(KPATH_SEPARATOR)));
     }
-    else
-    {
-        xdgdirList.clear();
-        xdgdirList.append(QString::fromLatin1("/etc/xdg"));
+
 #ifdef Q_WS_WIN
-        xdgdirList.append(installPath("kdedir") + QString::fromLatin1("etc/xdg"));
+    xdgdirList.append(installPath("kdedir") + QString::fromLatin1("etc/xdg"));
 #else
-        xdgdirList.append(QFile::decodeName(KDESYSCONFDIR "/xdg"));
+    xdgdirList.append(QFile::decodeName(KDESYSCONFDIR "/xdg"));
 #endif
-    }
+    xdgdirList.append(QString::fromLatin1("/etc/xdg"));
 
     QString localXdgDir = readEnvPath("XDG_CONFIG_HOME");
     if (!localXdgDir.isEmpty())
